packages/gatsby-image/src/index.js

```diff
@@ -69,6 +76,95 @@ function getIO() {
   return io
 }
 
+function generateImageSources(imageVariants) {
+  const initialImage = imageVariants[0]
+
+  if (imageVariants.length === 1) {
+    return initialImage.srcSetWebp ? (
+      <source
+        type={`image/webp`}
+        media={initialImage.media}
```

Comment:
Yeah it's a good question, I was thinking about an edge case where someone wanted to apply `media` to an image but not with multiple images? I can't imagine a legitimate use case for that though, happy to keep all `media` stuff with the multiple variants cases.

Reply:
I'm not against supporting it, it's not supported on the img element though, so wouldn't that require providing both sources for the first image too?

Reply:
It would, but we need to generate that initial source anyway if the user is using image variants. I'm happy to always generate that initial source regardless of if they're using variants or not (assuming no regressions on Safari). I had just assumed we'd want to keep the current no-`<source>` behavior for people not using variants though.